### ISSUE TITLE
Strange type inference with invalid return type

### ISSUE THREAD

**Issue by decorator-factory** [microsoft/pyright#2594]
```py
def aaa() -> callable:
    ...

bbb = aaa()

reveal_type(bbb)
```
This shows the type of `bbb` to be `(__obj: object) -> bool`. I think it should've been `Unknown` since `callable` isn't a valid type annotation.
 
This could happen if `aaa` and `bbb` are inside library code you don't have control over. It's tempting to use `callable` as a non-generic version of `typing.Callable`, so this is how I found this behaviour

**Comment by erictraut** [comment]
Yeah, I agree this should resolve to an "Unknown" type. This change will be in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.190, which I just published. It will also be included in the next release of pylance.
